core/package-graph/index.js

```diff
@@ -65,7 +66,26 @@ class PackageGraph extends Map {
           return currentNode.externalDependencies.set(depName, resolved);
         }
 
-        if (forceLocal || resolved.fetchSpec === depNode.location || depNode.satisfies(resolved)) {
+        let isLocal = forceLocal || resolved.fetchSpec === depNode.location || depNode.satisfies(resolved);
```

Comment:
Why not reuse `npm-package-arg`'s existing [`resolved.type === 'file'](https://github.com/npm/npm-package-arg#result-object)? Seems a lot easier than all of this complicated regular expression juggling.

Reply:
Ah yea that is a lot easier.

I was trying to solve a very specific case, but any local file is by definition a "local" dependency...